### DISCUSSION TITLE
Request ID not logged when used outside route handlers

### DISCUSSION THREAD

**Discussion by hrishikesh-k** [fastify/fastify#5045] [Q&A]
Hi!

Following the docs here: https://fastify.dev/docs/latest/Reference/Logging/#usage, specifically:

> You can trigger new logs outside route handlers by using the Pino instance from the Fastify instance

I'm doing something like:

```ts
import fastify from 'fastify'
import fastifyCookie from '@fastify/cookie'
const api = fastify({
  disableRequestLogging: true,
  logger: {
    base: {},
    level: 'debug',
    timestamp: false,
    transport: {
      options: {
        hideObject: true,
        messageFormat: '{reqId}: {msg}'
      },
      target: 'pino-pretty'
    }
  },
  requestIdHeader: 'x-nf-request-id'
})
api.register(fastifyCookie, {
  secret: {
    sign: stringToSign => {
      api.log.debug('hashing and encrypting cookie')
    },
    unsign: stringToUnsign => {
      api.log.debug('decrypting cookie')
    }
  }
})
```

Inside my route handlers, I get my logs as expected. However, the logs that I've shared in the example, do not log the request ID, they get logged as:

![image](https://github.com/fastify/fastify/assets/11403551/8aa0ae65-7801-49ea-99e6-3c5468d6ab20)

Is this expected? If not, what am I doing wrong?

**Comment by Eomm** [comment]
> Is this expected? If not, what am I doing wrong?

It is expected because `api.log` is called.
To get the reqId is necessary to call `request.log` instead.

In this case, I don't think you can do it unless a new feature request is implemented on fastify-cookie repo:
https://github.com/fastify/fastify-cookie/blob/master/plugin.js#L129-L138
